src/util/reference_processor.rs

```diff
@@ -236,6 +240,18 @@ impl ReferenceProcessor {
         e.trace_object(referent)
     }
 
+    /// This function is called when forwarding the references and referents (for MarkCompact). It
+    /// -   enqueues the referent if not yet reached, so that its children will be visited and
```

Comment:
When you say 'enqueue' here, do you mean the tracing queue? I would suggest avoiding using the word 'enqueue' here, as this method is actually used in a method named 'enqueue' which enqueues dead weak references to the reference queue. Just say it 'traces' the referent.